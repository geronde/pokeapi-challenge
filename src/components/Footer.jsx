import React from 'react'
import styled from 'styled-components';

const Wrapper = styled.footer`
    grid-area: footer;
    display: flex;
    align-items: center;
    justify-content: center;
`

const Footer = (props) => {
    return <Wrapper>This application was made for a challenge</Wrapper>

}

export default Footer;