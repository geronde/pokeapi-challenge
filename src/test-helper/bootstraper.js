import faker from 'faker';

export const generateTestPokemons = (n = 15, isType) => {
  const pokemons = [];

  for (let i = 0; i < n; i++) {
    let poke = { name: faker.random.word(), url: `fake-url/${i}/` };
    if (isType) {
      poke = { pokemon: { ...poke } };
    }
    pokemons.push(poke);
  }
  return pokemons;
};
