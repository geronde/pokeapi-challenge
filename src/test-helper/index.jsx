import { render as rtlRender } from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';

export const render = (ui, { ...renderOptions } = {}) =>{
    function Wrapper({ children }) {
      return <BrowserRouter>{children}</BrowserRouter>
    }
    return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
  }