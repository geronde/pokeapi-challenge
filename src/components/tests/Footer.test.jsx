import React from "react";
import {render} from '../../test-helper/index'
import Footer from '../Footer'

describe('Footer', () => {
    it('should display Footer correctly', () => {
       const { asFragment } = render(<Footer />)
       expect(asFragment()).toMatchSnapshot();
    });
});