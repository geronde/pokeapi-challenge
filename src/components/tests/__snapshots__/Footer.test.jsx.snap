// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should display Footer correctly 1`] = `
<DocumentFragment>
  .c0 {
  grid-area: footer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<footer
    class="c0"
  >
    This application was made for a challenge
  </footer>
</DocumentFragment>
`;
